### PR TITLE
Super File Deletion and BigQuery Connection Refactor

### PR DESCRIPTION
**Pull request by gestone**
Note: I have confirmed that all the tests pass and the code compiles, but I can't test that this all works on my local stack since it's still broken.

Will need someone to pull it down to test that I haven't somehow broken basic functionality.

### CODE REVIEW

File: soql-server-bq/src/main/scala/com/socrata/bq/query/BBQQuerier.scala [hunk 1]
```diff
@@ -2,45 +2,51 @@ package com.socrata.bq.query
...
+
+  // Whether or not batch queries should be executed. By default, batch queries
+  // should not be executed since queries should be performed immediately after the are issued rather
+  // than waiting around for idle resources.
+  // More information here: https://cloud.google.com/bigquery/querying-data?hl=en#batchqueries
+  private val BATCH : Boolean = false
 
-    run(projectId, queryString, batch, waitTime)
+  // BigQuery client
+  private val BIG_QUERY = {
```
Comment (george-socrata) [inline soql-server-bq/src/main/scala/com/socrata/bq/query/BBQQuerier.scala]: Bigquery is one word.

File: soql-server-bq/src/main/scala/com/socrata/bq/query/BBQQuerier.scala [hunk 2]
```diff
@@ -2,45 +2,51 @@ package com.socrata.bq.query
...
-    val bigquery: Bigquery = BBQServiceFactory.getService
-    val query: Job = asyncQuery(bigquery, projectId, queryString, batch)
-    val getRequest: Bigquery#Jobs#Get = bigquery.jobs.get(projectId, query.getJobReference.getJobId)
-    BBQUtils.pollJob(getRequest, waitTime)
-    val resultsRequest: Bigquery#Jobs#GetQueryResults = bigquery.jobs.getQueryResults(projectId, query.getJobReference.getJobId)
-    BBQUtils.getPages(resultsRequest)
+  @throws(classOf[Exception])
+  def query(queryString: String): Iterator[GetQueryResultsResponse] = {
+    val query: Job = asyncQuery(BIG_QUERY, projectId, queryString)
+    val getRequest: Bigquery#Jobs#Get = BIG_QUERY.jobs.get(projectId, query.getJobReference.getJobId)
```
Comment (george-socrata) [inline soql-server-bq/src/main/scala/com/socrata/bq/query/BBQQuerier.scala]: I've never seen this pound-symbol accessor before. Is it a java-in-scala specific operator?

Comment (george-socrata) [inline soql-server-bq/src/main/scala/com/socrata/bq/query/BBQQuerier.scala]: Oh wow, scala's implementation of nested classes is super funky.

**Comment by george-socrata** [comment]
I'd change `BIG_QUERY` to `BIGQUERY` or simply `bigquery`, but that's really up to you. Looks good.

**Comment by asapek** [comment]
Tested reading and writing, looks fine

**Comment by gestone** [comment]
@george-socrata will do, it's good to be consistent

**Pull request merged**